basket/news/backends/ctms.py

```diff
@@ -236,56 +246,65 @@ def to_vendor(data):
             value = value.strip()
         except AttributeError:
             pass
-        if value is None or value == "":
-            # TODO: Should we allow clearing previously set values?
+        # Skip empty values if new record or also unset in existing data
+        if (value is None or value == "") and not existing_data.get(name):
             continue
 
         # Place in CTMS contact structure
         if name in BASKET_TO_CTMS_NAMES:
             group_name, key = BASKET_TO_CTMS_NAMES[name]
             ctms_data.setdefault(group_name, {})[key] = value
+            if name in {"lang", "format"}:
```

Comment:
I haven't confirmed it for this code, but [others report](https://stackoverflow.com/a/58565258/10612) that `x in {"elem1", "elem2"}` is faster than `x in ["elem1", "elem2"]`, at least in Python 3. It would be faster if the set was initialized outside of the loop. It is probably insignificant for two items.

Reply:
I've now run it:

https://gist.github.com/jwhitlock/3ddaad9a06d387e54e2e1d530b248575

I show sets being faster than lists, and inline versions faster than defining outside the list, but they are all pretty close
```
function       time    factor
  set_in_loop  1.041s  (1.0x)
 set_out_loop  1.182s  (1.1x)
 list_in_loop  1.848s  (1.8x)
list_out_loop  2.019s  (1.9x)
```

Reply:
Nice! Looking forward to the blog post!